app/middleware/cache_headers.py

```diff
@@ -45,6 +51,9 @@ def __call__(self, request):
                 extra={"request": request}
             )
             patch_response_headers(response, settings.CACHE_MIDDLEWARE_SECONDS)
-            patch_cache_control(response, public=True)
+            directive = error_directives if response.status_code in (
+                502, 503, 504, 507
+            ) else public_directives
+            patch_cache_control(response, **directive)
```

Comment:
Patching here would not patch error on STATIC endpoints.

Reply:
@ltshb how do you want to patch static endpoints? in the original code they are not patched at all or everything but static is patched as follows.
```python
        elif (
            request.method in ('GET', 'HEAD') and
            not request.path.startswith(urlparse(settings.STATIC_URL).path)
        ):
            logger.debug(
                "Patching cache headers for request %s %s",
                request.method,
                request.path,
                extra={"request": request}
            )
            patch_response_headers(response, settings.CACHE_MIDDLEWARE_SECONDS)
            patch_cache_control(response, public=True)
```

Reply:
I meant with your previous code if an error occurred on a static endpoint it would not have set the cache control header falling back to CF default, now with your last commit error are patch regardless of the path.